contrib/babelfishpg_tsql/src/pltsql_coerce.c

```diff
@@ -885,6 +917,243 @@ run_tsql_best_match_heuristics(int nargs, Oid *input_typeids, FuncCandidateList
 	return new_candidates;
 }
 
+/*
+ * getImmediateBaseTypeOfUDT()
+ * This funciton 
+ */
+static Oid getImmediateBaseTypeOfUDT(Oid typeid)
+{
+	Relation					relsetting;
+	HeapTuple					tuple;
+	ScanKeyData					scankey;
+	SysScanDesc					scan;
+	bool						isnull;
+	Datum						datum;
+	Datum                       tsql_typename;
+	LOCAL_FCINFO(fcinfo, 1);
+
+	/* if common_utility_plugin_ptr is not initialised */
+	if (common_utility_plugin_ptr == NULL)
+		ereport(ERROR,
+				(errcode(ERRCODE_INTERNAL_ERROR),
+					errmsg("Failed to find common utility plugin.")));
+
+	/* if tsql_typename is NULL it implies that inputTypId corresponds to UDT */
+	InitFunctionCallInfoData(*fcinfo, NULL, 0, InvalidOid, NULL, NULL);
+	fcinfo->args[0].value = ObjectIdGetDatum(typeid);
+	fcinfo->args[0].isnull = false;
+	tsql_typename = (*common_utility_plugin_ptr->translate_pg_type_to_tsql) (fcinfo);
+
+	/* if given type is not an UDT then return InvalidOid */
+	if (tsql_typename)
+		return InvalidOid;
```

Comment:
Here function `getImmediateBaseTypeOfUDT()` is used to get the immediate base type of User Defined Types(UDT) only and for other types an InvalidOid is returned indicating the input type is not an UDT. @sumitj824 can you elaborate on why should we do this check before calling the function and not here ?, as removing this check from here will change the function itself from `getImmediateBaseTypeOfUDT()` to `getImmediateBaseType()`, which will return immediate base type of non-UDT datatypes as well, such as for input `nvarchar` it will return `varchar`. Is there any use-case of having such generic function to get immediate base type of non-UDT datatypes ?